src/Enterspeed.Source.UmbracoCms.V7/EventHandlers/MediaEventHandler.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Enterspeed.Source.UmbracoCms.V7.Contexts;
+using Enterspeed.Source.UmbracoCms.V7.Data.Models;
+using Enterspeed.Source.UmbracoCms.V7.Factories;
+using Umbraco.Core;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Umbraco.Core.Services;
+using Umbraco.Web;
+
+namespace Enterspeed.Source.UmbracoCms.V7.EventHandlers
+{
+    public class MediaEventHandler : ApplicationEventHandler
+    {
+        private static readonly EnterspeedJobFactory EnterspeedJobFactory = EnterspeedContext.Current.Services.JobFactory;
+
+        protected override void ApplicationStarted(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
+        {
+            MediaService.Saved += MediaService_Saved;
+            MediaService.Moved += MediaService_Moved;
+            MediaService.Trashed += MediaService_Trashed;
+        }
+
+        private static void MediaService_Trashed(IMediaService sender, MoveEventArgs<IMedia> e)
+        {
+            var isPublishConfigured = EnterspeedContext.Current.Services.ConfigurationService.IsPublishConfigured();
+
+            if (!isPublishConfigured)
+            {
+                return;
+            }
+
+            var entities = e.MoveInfoCollection.ToList();
+            var jobs = new List<EnterspeedJob>();
+
+            foreach (var mediaItem in entities.Select(ei => ei.Entity))
+            {
+                jobs.Add(EnterspeedJobFactory.GetDeleteJob(mediaItem, string.Empty, EnterspeedContentState.Publish));
+            }
+
+            EnqueueJobs(jobs);
+        }
+
+        private static void MediaService_Moved(IMediaService sender, MoveEventArgs<IMedia> e)
+        {
+            var isPublishConfigured = EnterspeedContext.Current.Services.ConfigurationService.IsPublishConfigured();
+
+            if (!isPublishConfigured)
+            {
+                return;
+            }
+
+            var entities = e.MoveInfoCollection.ToList();
+            var jobs = new List<EnterspeedJob>();
+
+            foreach (var mediaItem in entities.Select(ei => ei.Entity))
+            {
+                if (mediaItem.ContentType.Alias.Equals("Folder"))
```

Comment:
Could we move this into private method that can be reused accross these event handling?